x/concentrated-liquidity/internal/math/math.go

```diff
@@ -63,8 +71,9 @@ func CalcAmount0Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 	// additionally, without rounding, there exists cases where the swapState.amountSpecifiedRemaining.GT(sdk.ZeroDec()) for loop within
 	// the CalcOut/In functions never actually reach zero due to dust that would have never gotten counted towards the amount (numbers after the 10^6 place)
 	if roundUp {
-		return liq.Mul(diff).Quo(denom).Ceil()
+		return liq.Mul(diff).Quo(denom).RoundInt().ToDec()
```

Comment:
What was the reasoning behind ensuring safeness on this change?) I assumed ceiling would be safer then rounding)

Reply:
I was a little confused by how Ceil works. For a simplistic example, I would have a number like

10.999999

And using Ceil would return 10, which seems wrong.

Reply:
Do you have a branch with this test by any chance? `Ceil` seems to be functioning as intended on my end, and much of our rounding guardrails in GAMM rely on it.

In any case, `RoundInt` is not equivalent to rounding up, as it rounds down if the decimal is < .5

Reply:
Since this is orthogonal to the PR now (am no longer using the round up functionality) I will revert this and make an issue to investigate

Reply:
Thanks! Please revert this before merging

Reply:
Reverted https://github.com/osmosis-labs/osmosis/pull/4051/commits/6958a3d47f21635d1aa05858757f7ab18e3ae9b9